### ISSUE TITLE
CARGO_REGISTRY_TOKEN required for creating the pr

### ISSUE THREAD

**Issue by francisdb** [MarcoIeni/release-plz#1096]
## Bug description

I have dependabot set up on my project but since the `CARGO_REGISTRY_TOKEN` is not shared with dependabot the PR create fails.

`Run if [ -z "${CARGO_REGISTRY_TOKEN-unset}" ]`
see https://github.com/francisdb/vpin/actions/runs/6941969833/job/18883903400

Setting a dummy CARGO_REGISTRY_TOKEN for dependabot works around this

This workaround would not be needed if the `CARGO_REGISTRY_TOKEN` check would only be made in case of a release PR merge. But maybe the idea is that you want to be sure everything is set up fine before merging the PR?

- Would you like to work on a fix? [n]

**Comment by MarcoIeni** [comment]
I didn't get what's the issue with dependabot.
Release-plz requires the `CARGO_REGISTRY_TOKEN` to run `release-plz release` in CI.

Does https://github.com/MarcoIeni/release-plz-action/pull/96 help?

**Comment by francisdb** [comment]
> I didn't get what's the issue with dependabot.

When letting dependabot merge the it's dependency update pr by adding a comment to the dependabot issue, eg [@dpendabot squash and merge](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/managing-pull-requests-for-dependency-updates#managing-dependabot-pull-requests-with-comment-commands) the `release-pr` action will run as dependabot. But dependabot does not have access to the repository secrets. However for this step the secrets are not required.

Your fix should indeed solve the issue, thanks.

**Comment by francisdb** [comment]
Looks like this is still not fixed:
https://github.com/francisdb/vpin/actions/runs/7669146952/job/20902458480

If you do not set a command (default) both `release-pr` and `release` are run
During the `release` step no actual release is made since the current version already exists.
eg:
```
INFO  vpin 0.7.0: Already published - Tag v0.7.0 already exists
```
But the `CARGO_REGISTRY_TOKEN` is checked even if it's not going to be used.

**Comment by MarcoIeni** [comment]
thanks for reporting this again. I moved this check in the cli, just before running `cargo publish` 👍 
You can try again when I release the new version of the github action.
